internal/io/reader.go

```diff
@@ -79,3 +79,12 @@ func (yamlDtoReader) Read(reader io.Reader, v interface{}) errors.EdgeX {
 
 	return nil
 }
+
+func ReadAll(reader io.Reader) ([]byte, errors.EdgeX) {
+	bytes, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, errors.NewCommonEdgeX(errors.KindIOError, "AddEventRequest I/O reading failed", nil)
```

Comment:
modify the error message to general purpose

Reply:
moved this to `internal/core/data/controller/http/event.go`